forge/db/models/Project.js

```diff
@@ -12,7 +12,22 @@ module.exports = {
         id: { type: DataTypes.UUID, primaryKey: true, defaultValue: DataTypes.UUIDV4 },
         name: { type: DataTypes.STRING, allowNull: false },
         type: { type: DataTypes.STRING, allowNull: false },
-        url: { type: DataTypes.STRING, allowNull: false },
+        url: {
+            type: DataTypes.STRING,
+            allowNull: false,
+            get () {
+                const originalUrl = this.getDataValue('url').replace(/\/$/, '')
+                let httpAdminRoot = this.ProjectSettings?.[0]?.value.httpAdminRoot || this.ProjectTemplate?.settings?.httpAdminRoot
+                if (httpAdminRoot) {
+                    if (httpAdminRoot[0] !== '/') {
```

Comment:
Do we need this as well as the ProjectSettings controller validation earlier?

Reply:
I decided it safer to validate it in multiple places whilst I was figuring out who should 'own' the validation and cleaning.